commit f05ddd7fb0ad2ecf6a0f943e841a86108946a826
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-04-23T08:10:01-07:00 (6 months ago)
    
    TRITON-1420 Fabric nat ticket release in the provisioning workflow does not work correctly
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/workflows/add-nics.js b/lib/workflows/add-nics.js
index 71c0b1e..9389c40 100644
--- a/lib/workflows/add-nics.js
+++ b/lib/workflows/add-nics.js
@@ -23,7 +23,7 @@ var async;  // stub to keep jsl happy
 var common = require('./job-common');
 var fabricCommon = require('./fabric-common');
 
-var VERSION = '7.3.0';
+var VERSION = '7.3.1';
 
 
 /*
@@ -133,7 +133,7 @@ var workflow = module.exports = {
     },
 
     // If this was a fabric nat provision, clean up the ticket
-    fabricCommon.releaseTicketTask
+    fabricCommon.releaseFabricNatTickets
 
     ]),
     timeout: 300,
@@ -146,7 +146,7 @@ var workflow = module.exports = {
             modules: { sdcClients: 'sdc-clients', async: 'async' }
         },
         common.tasks.releaseVMTicketIgnoringErr,
-        fabricCommon.releaseTicketTask,
+        fabricCommon.releaseFabricNatTickets,
         {
             name: 'On error',
             body: function (job, cb) {
@@ -156,6 +156,6 @@ var workflow = module.exports = {
     ],
     oncancel: [
         common.tasks.releaseVMTicketIgnoringErr,
-        fabricCommon.releaseTicketTask
+        fabricCommon.releaseFabricNatTickets
     ]
 };
diff --git a/lib/workflows/fabric-common.js b/lib/workflows/fabric-common.js
index 53d95e6..7da46bf 100644
--- a/lib/workflows/fabric-common.js
+++ b/lib/workflows/fabric-common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -289,37 +289,6 @@ function provisionFabricNats(job, cb) {
 }
 
 
-/*
- * Release a fabric NAT ticket - intended to be called by the workflow that's
- * actually provisioning the NAT zone, not the one that called out to SAPI to
- * kick off the provision.
- */
-function releaseFabricTicket(job, cb) {
-    if (!job.params.ticket) {
-        return cb(null, 'No ticket to release');
-    }
-
-    var cnapi = new sdcClients.CNAPI({
-        url: cnapiUrl,
-        headers: { 'x-request-id': job.params['x-request-id'] }
-    });
-
-    cnapi.waitlistTicketRelease(job.params.ticket, function (err) {
-        if (err) {
-            if (err.code === 'ResourceNotFound') {
-                cb(null, 'Ticket released');
-            } else {
-                cb(err);
-            }
-
-            return;
-        }
-
-        cb(null, 'Released ticket ' + job.params.ticket);
-    });
-}
-
-
 /*
  * Wait for any pending fabric NAT provisions to complete.
  */
@@ -641,7 +610,6 @@ function releaseFabricNatTickets(job, cb) {
 module.exports = {
     acquireFabricTickets: acquireFabricTickets,
     provisionFabricNats: provisionFabricNats,
-    releaseFabricTicket: releaseFabricTicket,
     waitForFabricNatProvisions: waitForFabricNatProvisions,
 
     provisionChain: [
@@ -707,13 +675,5 @@ module.exports = {
         retry: 1,
         body: releaseFabricNatTickets,
         modules: { sdcClients: 'sdc-clients', async: 'async' }
-    },
-
-    releaseTicketTask: {
-        name: 'cnapi.release_fabric_nat_ticket',
-        timeout: 60,
-        retry: 1,
-        body: releaseFabricTicket,
-        modules: { sdcClients: 'sdc-clients' }
     }
 };
diff --git a/lib/workflows/provision.js b/lib/workflows/provision.js
index aea964b..8a53444 100644
--- a/lib/workflows/provision.js
+++ b/lib/workflows/provision.js
@@ -23,7 +23,7 @@ var nfsVolumes = require('./nfs-volumes');
 
 var wfapiUrl;
 
-var VERSION = '8.2.0';
+var VERSION = '8.2.1';
 
 
 /*
@@ -518,7 +518,7 @@ var workflow = module.exports = {
     },
 
     // If this was a fabric nat provision, clean up the ticket
-    fabricCommon.releaseTicketTask
+    fabricCommon.releaseFabricNatTickets
 
     ]),
     timeout: 3810,
@@ -567,7 +567,7 @@ var workflow = module.exports = {
     },
 
     // If this was a fabric nat provision, clean up the ticket
-    fabricCommon.releaseTicketTask,
+    fabricCommon.releaseFabricNatTickets,
     {
         name: 'On error',
         body: function (job, cb) {
@@ -590,6 +590,6 @@ var workflow = module.exports = {
     },
 
     // If this was a fabric nat provision, clean up the ticket
-    fabricCommon.releaseTicketTask
+    fabricCommon.releaseFabricNatTickets
     ]
 };
diff --git a/package.json b/package.json
index 3fe8ffd..90e755b 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.8.1",
+  "version": "9.8.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
